external/external.go

```diff
@@ -432,77 +395,62 @@ func (e *External) UpdateEndpoint(writer http.ResponseWriter, request *http.Requ
 //
 // The name of the Endpoint is derived from the URL the request was sent to and
 // the namespace comes from the daemon-wide configuration value.
-func (e *External) GetEndpoint(writer http.ResponseWriter, request *http.Request) {
+func (e *External) GetEndpoint(c echo.Context) error {
 	var (
 		endpoint string
-		ok       bool
-		v        = mux.Vars(request)
+		err      error
 	)
 
-	if endpoint, ok = v["name"]; !ok {
-		common.Error(writer, "missing endpoint name in the URL", http.StatusBadRequest)
-		return
+	endpoint = c.Param("name")
+	if endpoint == "" {
+		return &echo.HTTPError{
+			Code:     http.StatusBadRequest,
+			Internal: errors.New("missing endpoint name in the URL"),
+		}
 	}
 
 	log.Printf("GetEndpoint: getting info on endpoint %s", endpoint)
 
 	ept, err := e.EndpointController.Get(endpoint)
 	if err != nil {
-		common.Error(writer, err.Error(), http.StatusInternalServerError)
-		return
+		return err
 	}
 
 	log.Printf("GetEndpoint: done getting info on endpoint %s", endpoint)
 
-	WriteEndpoint(ept, writer)
+	returnOpts := &EndpointOptions{
+		Name:      ept.Name,
+		Namespace: ept.Namespace,
+		IP:        ept.Subsets[0].Addresses[0].IP,
+		Port:      ept.Subsets[0].Ports[0].Port,
+	}
 
-	log.Printf("GetEndpoint: done writing response for getting endpoint %s", endpoint)
+	return c.JSON(http.StatusOK, returnOpts)
 }
 
 // DeleteEndpoint is an http handler for deleting an Endpoints object from a k8s cluster.
 //
 // Expects no request body and returns no body in the response. Returns a 200
 // if you attempt to delete an Endpoints object that doesn't exist.
-func (e *External) DeleteEndpoint(writer http.ResponseWriter, request *http.Request) {
-	var (
-		endpoint string
-		ok       bool
-		v        = mux.Vars(request)
-	)
+func (e *External) DeleteEndpoint(c echo.Context) error {
+	var endpoint string
 
-	if endpoint, ok = v["name"]; !ok {
-		common.Error(writer, "missing endpoint name in the URL", http.StatusBadRequest)
-		return
+	endpoint = c.Param("name")
+	if endpoint == "" {
+		return &echo.HTTPError{
+			Code:     http.StatusBadRequest,
+			Internal: errors.New("missing endpoint name in the URL"),
+		}
 	}
 
 	log.Printf("DeleteEndpoint: deleting endpoint %s", endpoint)
 
 	err := e.EndpointController.Delete(endpoint)
 	if err != nil {
-		common.Error(writer, err.Error(), http.StatusInternalServerError)
-		return
+		log.Error(err) // Repeated Deletion requests shouldn't return errors.
```

Comment:
Are there any other types of errors that might be returned? If so, is there a way to distinguish an attempt to delete a non-existent endpoint from other types of errors?

Reply:
I haven't figured that out. I'm not even sure that this function call is needed anymore, to be honest.